src/rubyfmt.rb

```diff
@@ -2477,17 +2551,27 @@ def on_string_literal(*args, &blk)
           next if part[1].nil?
           case part[1][0]
           when :@tstring_content
-            fixup_tstring_content_for_double_quotes(part[1][1])
+            fixup_tstring_content_for_double_quotes(
+              part[1][1],
+              is_single_quote: true,
+              delimiter: quote,
+            )
           else
             raise "got non tstring content in single string"
```

Comment:
Do we need to guard against this? Without a serious Ripper bug, I'm not sure how we could get a `string_embexpr` in the wrong type of string. If we don't need this guard clause, we can probably collapse the `/'/` and `/\W/` branches into one.

Reply:
This is definitely a "is the ruby parser well written" test, but I'd like to keep it in case ruby ever does weird things in the future.